dmci/api/worker.py

```diff
@@ -174,39 +174,51 @@ def distribute(self):
             The names of all distributors that were skipped (invalid)
         failed_msg : list of str
             The messages returned by the failed jobs
+        retry_status : bool
+             True if any one distributer fails to process or not available
+        retry : list of str
+            The name of all distributers that were not able process or not available
         """
         status = True
         valid = True
+        retry_status = False
         called = []
         failed = []
+        retry = []
         skipped = []
         failed_msg = []
 
         for dist in self._conf.call_distributors:
-            if dist not in self.CALL_MAP:
-                skipped.append(dist)
-                continue
-            obj = self.CALL_MAP[dist](
-                self._dist_cmd,
-                xml_file=self._dist_xml_file,
-                metadata_UUID=self._dist_metadata_id_uuid,
-                worker=self,
-                path_to_parent_list=self._kwargs.get(
-                    "path_to_parent_list", None),
-            )
-            valid &= obj.is_valid()
```

Comment:
I don't understand why we have used these obscure operators that only serve to hide the intention of the code but as far as I understand `valid` is assigned the value `True`/`False` if `obj.valid()` returns `True`/`False`. In this case, line 198 (or 205 in your version) should just be `if valid:`

Reply:
in this case ,if any of the 'obj is not valid ,the 'valid, variable is set to false.This is just used for logging  purpose.

 But we need the validity of the current 'obj' to decide whether to call or not call 'obj.run()'. So the line 198 should be as it is now.

Reply:
aha - ok